src/libs/migrations/PersonalDetailsByAccountID.js

```diff
@@ -0,0 +1,208 @@
+                    });
+
+                    if (whisperedToAccountIDs.length === reportAction.whisperedTo.length) {
+                        newReportAction.whisperedToAccountIDs = whisperedToAccountIDs;
+                    } else {
+                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because whisperedToAccountIDs not found`);
+                        return;
+                    }
+                }
+
+                if (lodashHas(reportAction, ['childOldestFourEmails']) && !lodashHas(reportAction, ['childOldestFourAccountIDs'])) {
+                    reportActionsWereModified = true;
+                    const childOldestFourEmails = reportAction.childOldestFourEmails.split(',');
+                    const childOldestFourAccountIDs = [];
+                    _.each(childOldestFourEmails, (login) => {
+                        const accountID = _.get(oldPersonalDetails, [login.trim(), 'accountID']);
+                        if (accountID) {
+                            childOldestFourAccountIDs.push(accountID);
+                        }
+                    });
+
+                    if (childOldestFourAccountIDs.length === childOldestFourEmails.length) {
+                        newReportAction.childOldestFourAccountIDs = childOldestFourAccountIDs.join(',');
+                    } else {
+                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because childOldestFourAccountIDs not found`);
+                        return;
+                    }
+                }
+
+                if (lodashHas(reportAction, ['originalMessage', 'participants']) && !lodashHas(reportAction, ['originalMessage', 'participantAccountIDs'])) {
+                    reportActionsWereModified = true;
+                    const participantAccountIDs = [];
+                    _.each(reportAction.originalMessage.participants, (participant) => {
+                        const participantAccountID = _.get(oldPersonalDetails, [participant, 'accountID']);
+                        if (participantAccountID) {
+                            participantAccountIDs.push(participantAccountID);
+                        }
+                    });
+
+                    if (participantAccountIDs.length === reportAction.originalMessage.participants.length) {
+                        newReportAction.originalMessage.participantAccountIDs = participantAccountIDs;
+                    } else {
+                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because originalMessage.participantAccountIDs not found`);
+                        return;
+                    }
+                }
+
+                newReportActionsForReport[reportActionID] = newReportAction;
+            });
+
+            // Only include the reportActions from this report if they were modified in any way
```

Comment:
Just to be clear... If any reportAction in a report has the accountID data... we'll skip adding it to `newReportActionsForReport`, which means we'll remove it from Onyx?

Migrations happen BEFORE AppInit right? So this is only for existing Onyx data when the app refreshes, right?

Reply:
Ah, not quite. Basically if we are leaving the reportActions for a given report untouched, then we won't add them here, because that would make us unnecessarily do Onyx.set for a reportActions object that hasn't changed .

If there were any reportActions for this report that were changed (either by updating the reportAction or removing it) then we want to include it so that it will be included in the Onyx.set.

Reply:
aah ok we're only going to skip adding reportActions to `newReportActionsForReport` iff those report actions don't have successful migrations (using accountIDs from `personalDetails) - so they'll effectively be removed from onyx

And now i see the optimization you're talking about too 👍 thanks!!

Reply:
No problem! Let me know if you think I can make that clearer somehow.

Reply:
Maaaybe this is more clear / more accurate? Let me know what you think

```suggestion
            // Only include the reportActions from this report if at least one reportAction in this report
            // was modified in any way.
```